scraper/scraperhelper/obs_logs_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package scraperhelper
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/testdata"
+	"go.opentelemetry.io/collector/scraper"
+)
+
+func TestScrapeLogsDataOp(t *testing.T) {
+	tt, err := componenttest.SetupTelemetry(receiverID)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
+
+	parentCtx, parentSpan := tt.TelemetrySettings().TracerProvider.Tracer("test").Start(context.Background(), t.Name())
+	defer parentSpan.End()
+
+	params := []testParams{
+		{items: 23, err: partialErrFake},
+		{items: 29, err: errFake},
+		{items: 15, err: nil},
+	}
+	for i := range params {
+		var sf scraper.ScrapeLogsFunc
+		sf, err = newObsLogs(func(context.Context) (plog.Logs, error) {
+			return testdata.GenerateLogs(params[i].items), params[i].err
+		}, receiverID, scraperID, tt.TelemetrySettings())
+		require.NoError(t, err)
+		_, err = sf.ScrapeLogs(parentCtx)
+		require.ErrorIs(t, err, params[i].err)
+	}
+
+	spans := tt.SpanRecorder.Ended()
+	require.Equal(t, len(params), len(spans))
+
+	var scrapedLogRecords, erroredLogRecords int
+	for i, span := range spans {
+		assert.Equal(t, "scraper/"+scraperID.String()+"/ScrapeLogs", span.Name())
+		switch {
+		case params[i].err == nil:
+			scrapedLogRecords += params[i].items
+			require.Contains(t, span.Attributes(), attribute.KeyValue{Key: scrapedLogRecordsKey, Value: attribute.Int64Value(int64(params[i].items))})
```

Comment:
Use `attribute.Int64` instead of manually construct the key and the value?

Reply:
Updated in https://github.com/open-telemetry/opentelemetry-collector/pull/12036/commits/dde2390a2b83a9ab9f4951a7348e906c511d9638, also raised #12078 for obs metrics test.